### PR TITLE
feat(docs): add troubleshooting guide

### PR DESCRIPTION
**Pull request by dhmlau**
With the new acceptance criteria in https://github.com/strongloop/loopback-next/issues/5701, I'm taking a slightly different approach and making this PR as the first one.
- add the troubleshooting guide in the sidebar
- add "running and debugging app" and "setting debug strings" under it. 

Screen Shot 2020-06-15 at 9 56 24 PM

Related to https://github.com/strongloop/loopback-next/pull/5705. 

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: docs/site/Troubleshooting-guide.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+layout: page
+keywords:
+  LoopBack 4.0, LoopBack 4, Node.js, TypeScript, OpenAPI, Troubleshooting
+tags:
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Troubleshooting-guide.html
+summary: Transactional API usage in LoopBack 4
+---
+
+The following documentation pages can help you troubleshoot problems you
```
Comment (hacksparrow) [inline docs/site/Troubleshooting-guide.md:13]: "A list of troubleshooting guides for commonly encountered problems."?

**Comment by bajtos** [comment]
Thank you @dhmlau for starting this effort!

It makes me wonder if we can align the troubleshooting guide with the new documentation system (#5549). I think a lot of the content is (and will be) in "how-to guide" style, in which case it should go to "Usage scenarios". Except nobody would look for troubleshooting guides there. Should we perhaps rename "Usage scenarios" to "How-to guides" first, and then nest troubleshooting guides under "How-to guides"?

**Comment by agnes512** [comment]
> I think a lot of the content is (and will be) in "how-to guide" style, in which case it should go to "Usage scenarios". 

+1

**Comment by dhmlau** [comment]
> Should we perhaps rename "Usage scenarios" to "How-to guides" first, and then nest troubleshooting guides under "How-to guides"?

+1 on renaming. 

I'm not sure about moving the troubleshooting guides under "how-to guides".  To me, the "how-to guides" helps the users to do certain thing; and the "troubleshooting guide" is used when something goes wrong. I guess we may not want to mix them up? 

Having said that, we can find the instruction for running a LB application a better home. Probably not "how-tos" either, because that's probably very first few things users need to know.

**Comment by agnes512** [comment]
Maybe we can also rename the guide to something like "How to troubleshoot when something goes wrong" x)

**Comment by bajtos** [comment]
Thank you for thoughtful comments.

I agree to keep "Troubleshooting" as the top-level sidebar entry then. Can we somehow document that these guides should be practical (describe how to troubleshoot a specific problem) and that content explaining troubleshooting-related content should go elsewhere?

File: docs/site/sidebars/lb4_sidebar.yml [hunk 2]
```diff
@@ -807,6 +799,19 @@ children:
     url: Deploy-for-GDPR-readiness.html
     output: 'web, pdf'
 
+- title: 'Troubleshooting Guide'
```
Comment (bajtos) [inline docs/site/sidebars/lb4_sidebar.yml:802]: Personally, I would use a shorter title: "Troubleshooting", but I am ok with the longer version too.

If we decide to keep the longer form: Considering there are multiple pages nested under this entry, I think we should use plural (Troubleshooting Guides).

File: docs/site/Troubleshooting-guide.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+encountered.
+
+- [Running and debuggin apps](Running-and-debugging-apps.md)
+- [Setting debug strings](Setting-debug-strings.md)
```
Comment (bajtos) [inline docs/site/Troubleshooting-guide.md:17]: Do we want the manual process of keeping this list up to date? I am will be one of the first people to forget to update it.

Is it possible to generate this list automatically? I vaguely remember that I saw a markdown page using Jekyll syntax to enumerate the list of child nodes, but I cannot find it now.

Do we actually need to list the pages here at all? I mean we already have a list of child nodes in the left sidebar. Is it perhaps possible to remove this page entirely and keep the "Troubleshooting Guide" item in the sidebar as an expander (a group name) only?

File: docs/site/sidebars/lb4_sidebar.yml (hunk 2 above)
Comment (dhmlau) [reply docs/site/sidebars/lb4_sidebar.yml:802]: I like the shorter title 👍

File: docs/site/Troubleshooting-guide.md (hunk 3 above)
Comment (dhmlau) [reply docs/site/Troubleshooting-guide.md:17]: > Is it possible to generate this list automatically? I vaguely remember that I saw a markdown page using Jekyll syntax to enumerate the list of child nodes, but I cannot find it now.

Let me find out. 

> Do we actually need to list the pages here at all? I mean we already have a list of child nodes in the left sidebar. Is it perhaps possible to remove this page entirely and keep the "Troubleshooting Guide" item in the sidebar as an expander (a group name) only?

I vaguely remember Jekyll expects each entry in the sidebar to have the `url`, i.e. need a page for each entry. But I can double check.

**Comment by dhmlau** [comment]
Closing in favor of #6371

**Pull request closed without merging**